apps/test/unit/lib/ui/accounts/AddPasswordFormTest.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {mount} from 'enzyme';
+import sinon from 'sinon';
+import {expect} from '../../../../util/configuredChai';
+import AddPasswordForm, {SAVING_STATE, SUCCESS_STATE, PASSWORDS_MUST_MATCH} from '@cdo/apps/lib/ui/accounts/AddPasswordForm';
+
+describe('AddPasswordForm', () => {
+  let wrapper, handleSubmit;
+
+  beforeEach(() => {
+    handleSubmit = () => {};
+    wrapper = mount(
+      <AddPasswordForm
+        handleSubmit={handleSubmit}
+      />
+    );
+  });
+
+  it('enables form submission if passwords are present and match', () => {
+    wrapper.setState({
+      password: 'mypassword',
+      passwordConfirmation: 'mypassword'
+    });
+    const submitButton = wrapper.find('Button');
+    expect(submitButton).not.to.have.attr('disabled');
+  });
+
+  it('disables form submission if passwords are empty', () => {
+    wrapper.setState({
+      password: '',
+      passwordConfirmation: ''
+    });
+    expect(wrapper.find('Button')).to.have.attr('disabled');
+  });
+
+  it('disables form submission if passwords do not match', () => {
+    wrapper.setState({
+      password: 'newpassword',
+      passwordConfirmation: 'notnewpassword'
+    });
+    expect(wrapper.find('Button')).to.have.attr('disabled');
+  });
+
+  it('renders a password mismatch validation error if passwords do not match', () => {
+    wrapper.setState({
+      password: 'newpassword',
+      passwordConfirmation: 'notnewpassword'
+    });
+    expect(wrapper.find('FieldError')).to.have.text(PASSWORDS_MUST_MATCH);
+  });
+
+  it('renders the form submission state', () => {
+    wrapper.setState({
+      submissionState: SAVING_STATE
+    });
+    expect(wrapper.find('#uitest-add-password-status')).to.have.text(SAVING_STATE);
+  });
+
+  describe('on successful submission', () => {
+    beforeEach(() => {
+      handleSubmit = sinon.stub().resolves({});
+      wrapper = mount(
+        <AddPasswordForm
+          handleSubmit={handleSubmit}
+        />
+      );
+      wrapper.setState({
+        password: 'mypassword',
+        passwordConfirmation: 'mypassword'
+      });
+      const submitButton = wrapper.find('Button');
+      submitButton.simulate('click');
+    });
+
+    afterEach(() => {
+      handleSubmit.resetBehavior();
+    });
+
+    it('resets the password field to its default state', async () => {
+      await handleSubmit;
```

Comment:
`await handleSubmit` is common to every case in this `describe` block.  Move it into the `beforeEach`? (which can also be async)